client/tests/integration/extra_functional/connected_peers.rs

```diff
@@ -57,7 +57,7 @@ fn register_new_peer() -> Result<()> {
         .1
         .submit_blocking(register_peer)?;
     peer_clients.push((&new_peer, Client::test(&new_peer.api_address)));
-    thread::sleep(pipeline_time * 2); // Wait for some time to allow peers to connect
+    thread::sleep(pipeline_time * 2 * 20); // Wait for some time to allow peers to connect
 
     check_status(&peer_clients, 2);
```

Comment:
What if poll status periodically instead of waiting for a fixed amount of time?

Reply:
Waiting mechanisms is what was done in #5013 so instead on relying on `sleep`s or fixed time polling we listen for events from iroha.
So i think we can postpone this change and merge this PR because it indeed reduce flakiness of `extra_functional` tests.